include/fmt/compile.h

```diff
@@ -11,7 +11,15 @@
 #include "format.h"
 
 FMT_BEGIN_NAMESPACE
-namespace detail {
+FMT_MODULE_EXPORT_BEGIN
+
+// compile-time support
+namespace ct {
```

Comment:
Why new namespace?

Reply:
This is stuff that was in namespace `detail` before. It must be made visible to be used from importers, f.e. by the `""_cf` udl. The `compiled_string` tag class can be moved into namespace `fmt` just as well.

Reply:
Why can't we make `detail::compiled_string` visible? It's an implementation detail and not supposed to be directly used by anyone, so `detail` is the correct namespace.

Reply:
If we make `detail::compiled_string` visible then the namespace `detail` becomes implicitly visible, too. This disenfranchises testing from ever checking if anything from that namespace becomes unwittingly visible that was never meant to be visible int the first place. This is the sole reason why I introduced namespace `ct`: everything in there is the sole minimum of non-public API that needs to be visible. I'd be sad if we'd call namespace `detail` "open season" to visibility. I was bugging Cameron to fix the current issue in msvc of `detail` being visible even if nothing in there is currently visible (he checked this with his BMI explorer tool) and he told me that this should be easy to fix. 🤞

I've just finished reworking the PR and peeked here to check for any news. Part of the rework was moving `compiled_string` to namespace `fmt` because I got the feeling you were objecting to putting it into `ct` where other necessarily visible compile-time related stuff is moved to as well. I will push the PR as it is now. Changing it in either direction is a piece of cake and can be done within a few minutes.

Reply:
Not making `detail` visible makes sense but `ct` looks like part of the public API which is undesirable. I suggest renaming to something like `detail_exported` to make it clear that this is an internal API but exported for module purposes.

Reply:
Done